packages/next-acms/tests/test-api-compatibility.test.ts

```diff
@@ -0,0 +1,75 @@
+import { expect } from '@jest/globals';
+import { DrupalClient } from 'next-drupal';
+import { testApiCompatibility } from '../src/test-api-compatibility';
+
+const createMockClient = (contentTypes: Array<string>) => {
+  const links = {};
+  contentTypes.forEach((contentType) => {
+    links[contentType] = { href: `http://🐈/jsonapi/${contentType}` };
+  });
+
+  return jest
+    .spyOn(DrupalClient.prototype, 'getIndex')
+    .mockImplementation(() => {
+      return Promise.resolve({
+        jsonapi: {
+          version: '1.0',
+          meta: [],
+        },
+        meta: {
+          count: 0,
+        },
+        data: [],
+        errors: [],
+        links: links,
+      });
+    });
+};
+
+afterEach(() => {
+  jest.restoreAllMocks();
+});
+
+describe('testApiCompatibility', () => {
+  test('it should not throw for identical data model', async () => {
+    const contentTypes = ['node--test'];
+    createMockClient(contentTypes);
+    const client = new DrupalClient('http://🐈');
+    await testApiCompatibility(contentTypes, client);
+    expect(client.getIndex).toBeCalledTimes(1);
+  });
+
+  test('it should throw for invalid data model', async () => {
+    createMockClient([]);
+    const client = new DrupalClient('http://🐈');
+    await expect(
+      async () => await testApiCompatibility(['node--test'], client),
+    ).rejects.toThrow('Content type node--test does not exist in the backend.');
+    expect(client.getIndex).toBeCalledTimes(1);
+  });
+
+  test('it should not throw for content type not supported by client', async () => {
```

Comment:
Should the name of the test be this instead since `node--test` supported by the client?

```suggestion
  test('it should not throw for content type supported by client', async () => {
```

Reply:
Ah I see, makes sense now